.github/workflows/pre-build_checks.yaml

```diff
@@ -45,26 +45,39 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
       with:
         fetch-depth: 1
     - name: Installing python
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
       with:
         python-version: 3.x
+        cache: pip
+    - name: Install graphviz
+      run: |
+        sudo apt-get update
+        sudo apt-get install graphviz
     - name: Generate documentation
       run: |
         cd doc/sphinx
         pip install -r requirements.txt
         mkdir source/_static
         ./extract_rst.py
+        # Test building for the homepage
         sphinx-build -W -b json -d build/doctrees source build/json
+        # This is suitable for user testing
+        ./make.py html -graphs
+    - name: Upload artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: Widelands_Developer_Documentation_${{ github.sha }}
```

Comment:
You may want to rename this to `Widelands-${{ github.sha }}-Developer-Documentation`.
Then it should be picked up by the automated development releases (Pattern `Widelands-<hash>-<os matrix>`). Otherwise the artifact is only available for logged in users.

Reply:
Thanks for the tip! :)